src/cpp/src/llm_pipeline.cpp

```diff
@@ -90,21 +121,103 @@ ov::genai::EncodedResults multinominal_decoding(
 
 EncodedResults beam_search(ov::InferRequest& lm, ov::Tensor prompts, ov::Tensor attention_mask, GenerationConfig config);
 
-
-class LLMPipeline::LLMPipelineImpl {
+class LLMPipelineImplBase {
 public:
-    ov::InferRequest m_model_runner;
+    LLMPipelineImplBase(const Tokenizer& tokenzer,
+                        const GenerationConfig& config = {},
+                        const std::string& chat_template = "");
+
+    virtual DecodedResults generate(
+        StringInputs inputs,
+        OptionalGenerationConfig generation_config,
+        StreamerVariant streamer
+    ) = 0;
+
+    virtual EncodedResults generate(
+        const EncodedInputs& inputs,
+        OptionalGenerationConfig generation_config,
+        StreamerVariant streamer
+    ) = 0;
+
+    virtual void start_chat() = 0;
+    virtual void finish_chat() = 0;
+
+    std::string apply_chat_template(const std::vector<std::pair<std::string, std::string>>& prompts) const;
```

Comment:
In the latest version we moved `apply_chat_template` to `ov::genai::Tokenizer` to make it similar to HF. Unfortunately the latest version is in release and not cherry-picked to release.

Reply:
That makes things easier, I'll rebase once changes are available, thank you!

Reply:
Also removed, even less changes now!